chain/client/src/client_actor.rs

```diff
@@ -257,364 +263,451 @@ impl Actor for ClientActor {
     }
 }
 
-impl Handler<WithSpanContext<NetworkClientMessages>> for ClientActor {
-    type Result = NetworkClientResponses;
-
-    #[perf]
-    fn handle(
+impl ClientActor {
+    fn wrap<Req: std::fmt::Debug + actix::Message, Res>(
```

Comment:
This one here doesn't spark joy, but it seems that the first-best way to get rid of this is to remove actix. 

Another alternative is intead of 

```rust
self.wrap(msg,ctx,"NetworkAdversarialMessage",|this,msg| match msg {
```

to do 

```
let _span = self.message_span(msg, ctx, "NetworkAdversarialMessage");
match msg
```

I think we could replace closure with guard here? 

No strong opinion though. 


Only, what's up with `,` there? Seems like a round of `cargo fmt` is in order?

Reply:
this would be a multi-purpose guard keeping span, timer and delay detector. I don't think it will be more readable.